### PR TITLE
Registries

### PR DESCRIPTION
**Pull request by ajski**
## Purpose

Add new test for Registries.

## Summary of Changes

Add new test for Registries landing and discover page.  Add test for registration detail page. Rename RegistriesPage to RegistriesLandingPage.

## Ticket

https://openscience.atlassian.net/browse/QA-83

### CODE REVIEW

File: pages/registries.py [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
+
+    identity = Locator(By.CSS_SELECTOR, '[data-test-share-logo]')
+    loading_indicator = Locator(By.CSS_SELECTOR, '.ball-scale')
+
+    # Group Locators
+    search_results = GroupLocator(By.CSS_SELECTOR, '._SearchResult_10ty34')
+
+class RegistrationDetailPage(GuidBasePage):
+    url_base = urljoin(settings.OSF_HOME, '{guid}')
+    identity = Locator(By.CSS_SELECTOR, '[data-test-result-title-id]')
```
Comment (laurenrevere) [inline pages/registries.py]: This identity doesn't seem right. That looks like a selector from an embosfed page. The registration detail page isn't emberized.

File: tests/test_navbar.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from pages.register import RegisterPage
 from pages.project import MyProjectsPage
 from pages.dashboard import DashboardPage
-from pages.registries import RegistriesPage
+from pages.registries import RegistriesLandingPage
```
Comment (laurenrevere) [inline tests/test_navbar.py]: Good catch!

File: pages/registries.py [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
+    url = settings.OSF_HOME + '/registries/discover'
+
+    identity = Locator(By.CSS_SELECTOR, '[data-test-share-logo]')
+    loading_indicator = Locator(By.CSS_SELECTOR, '.ball-scale')
+
+    # Group Locators
+    search_results = GroupLocator(By.CSS_SELECTOR, '._SearchResult_10ty34')
+
+class RegistrationDetailPage(GuidBasePage):
+    url_base = urljoin(settings.OSF_HOME, '{guid}')
```
Comment (laurenrevere) [inline pages/registries.py]: For `GuidBasePage`, the default url is OSF_HOME + guid, so you actually don't need to redefine `url_base`.

File: pages/registries.py [hunk 4]
```diff
@@ -1,16 +1,34 @@
...
 from components.navbars import RegistriesNavbar
-from pages.base import OSFBasePage
+from pages.base import OSFBasePage, GuidBasePage
 
 
-class RegistriesPage(OSFBasePage):
+class BaseRegistriesPage(OSFBasePage):
+
+    # Components
+    navbar = ComponentLocator(RegistriesNavbar)
```
Comment (laurenrevere) [inline pages/registries.py]: Nice class usage to store common navbar component!

File: tests/test_registries.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+@pytest.fixture
+def landing_page(driver):
+    landing_page = RegistriesLandingPage(driver)
+    landing_page.goto()
+    return landing_page
+
+class TestRegistriesLandingPage:
+    @markers.smoke_test
+    @markers.core_functionality
+    def test_landing_page_loads(self, driver):
```
Comment (laurenrevere) [inline tests/test_registries.py]: You can leave it if you want, but for the sake of keeping the suite as lean as possible I would remove this. Any other test that includes the landing page will check it loads by default.

File: tests/test_registries.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        RegistriesLandingPage(driver).goto()
+
+class TestRegistriesDiscoverPage:
+
+    @markers.core_functionality
+    def test_discover_page_loads(self, driver):
```
Comment (laurenrevere) [inline tests/test_registries.py]: I would remove this as well.

File: tests/test_registries.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    @markers.smoke_test
+    @markers.core_functionality
+    def test_search_results_exist(self, driver, landing_page):
+        landing_page.search_button.click()
+        discover_page = RegistriesDiscoverPage(driver, verify=True)
+        discover_page.loading_indicator.here_then_gone()
+        assert len(discover_page.search_results) > 0
+
+    def test_detail_page(self, driver):
```
Comment (laurenrevere) [inline tests/test_registries.py]: This test looks good, but I'm not sure it actually tests what we said it tests based on the locators above.

File: pages/registries.py (hunk 3 above)
Comment (ajski) [reply pages/registries.py]: Done!

File: pages/registries.py [hunk 8]
```diff
@@ -37,16 +36,11 @@ def get_first_non_withdrawn_registration(self):
...
 
-
-class EmberRegistrationDetailPage(GuidBasePage):
-    url_base = urljoin(settings.OSF_HOME, '{guid}')
-    identity = Locator(By.CSS_SELECTOR, '[data-test-registration-title]')
-    title = Locator(By.CSS_SELECTOR, '[data-test-registration-title]')
-
-
 class RegistrationDetailPage(GuidBasePage):
-    waffle_override = {'ember_registries_detail_page': EmberRegistrationDetailPage}
```
Comment (ajski) [inline pages/registries.py]: We removed waffle_override because the Registries detail overview page has been 'emberized', so it's no longer needed.  ~Josh

**Pull request merged**